core/jmx/impl/src/main/java/org/opennms/netmgt/jmx/impl/connection/connectors/Jsr160ConnectionFactory.java

```diff
@@ -72,21 +85,36 @@ public static JmxServerConnectionWrapper getMBeanServerConnection(Map<String,?>
         String urlPath  = ParameterMap.getKeyedString( propertiesMap, "urlPath",  "/jmxrmi");
         String username = ParameterMap.getKeyedString(propertiesMap, "username", null);
         String password = ParameterMap.getKeyedString(propertiesMap, "password", null);
+        long timeout    = ParameterMap.getKeyedLong(propertiesMap, "timeout", DEFAULT_TIMEOUT);
         
-        Jsr160ConnectionWrapper connectionWrapper = null;
-        JMXServiceURL url = null;
+        //Jsr160ConnectionWrapper connectionWrapper = null;
         
         final String hostAddress = InetAddressUtils.str(address);
         LOG.debug("JMX: {} - service:{}//{}:{}{}", factory, protocol, hostAddress, port, urlPath);
 
         if (factory == null || factory.equals("STANDARD")) {
-                url = getUrl(address, port, protocol, urlPath);
+                final JMXServiceURL url = getUrl(address, port, protocol, urlPath);
                 
                 // Connect a JSR 160 JMXConnector to the server side
-                JMXConnector connector = JMXConnectorFactory.connect(url);
-                MBeanServerConnection connection = connector.getMBeanServerConnection();
-                
-                return new Jsr160ConnectionWrapper(connector, connection);
+
+                Callable<JMXConnector> task = new Callable<JMXConnector>() {
+                    public JMXConnector call() throws IOException {
+                       return JMXConnectorFactory.connect(url);
+                    }
+                 };
+                 Future<JMXConnector> future = executor.submit(task);
+                 JMXConnector connector = null;
+                 try {
+                     connector = future.get(timeout, TimeUnit.MILLISECONDS);
+                     MBeanServerConnection connection = connector.getMBeanServerConnection();
+                     return new Jsr160ConnectionWrapper(connector, connection);
+                } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                    LOG.info("Exception connecting JMXConnectorFactory url {} , Error: {}", url, e.getMessage());
+                    throw new ConnectException("Error connecting JMXConnectionFactory  " + url);
+                } finally {
+                    future.cancel(true);
```

Comment:
Sorry, I missed this. We'll add the missing log statements, there's no reason not to.